src/vdaf/prio3.rs

```diff
@@ -273,6 +273,19 @@ where
         self.typ.verifier_len()
     }
 
+    fn derive_joint_randomness<'a>(parts: impl Iterator<Item = &'a Seed<L>>) -> Seed<L> {
```

Comment:
Very nice implementation! Quick question: Because I'm unfamiliar with this use of `impl`, I would have written something like
```rust
fn derive_joint_randomness<'a, I: Iterator<Item = &'a Seed<L>>, const L: usize>(parts: I) -> Seed<L> {
   ...
}
```
Obviously this is way uglier! My question is if this is the problem `impl` was designed to solve here. Or perhaps what I'm describing is not actually grammatically correct, or hides some inefficiency?

Reply:
`impl Trait` syntax in argument position is roughly equivalent to an explicit type parameter with that same bound, with the exception that you can't explicitly specify a type with a turbofish, (like `derive_joint_randomness::<std::slice::Iter<'_, &'_ Seed<16>>>(parts)`) since there isn't a real type parameter.

Reply:
Thanks!